fdbserver/workloads/DataDistributionMetrics.actor.cpp

```diff
@@ -26,52 +26,110 @@
 		return Standalone<StringRef>(format("Value/%08d", deterministicRandom()->randomInt(0, 10e6)));
 	}
 
-	ACTOR static Future<Void> _start(Database cx, DataDistributionMetricsWorkload* self) {
-		state int tNum;
-		for (tNum = 0; tNum < self->numTransactions; ++tNum) {
-			loop {
-				state ReadYourWritesTransaction tr(cx);
-				try {
-					state int i;
-					for (i = 0; i < self->writesPerTransaction; ++i) {
-						tr.set(StringRef(format("Key/%08d", tNum * self->writesPerTransaction + i)), getRandomValue());
+	Key keyForIndex(int n) { return doubleToTestKey((double)n / nodeCount, keyPrefix); }
+
+	ACTOR static Future<Void> ddRWClient(Database cx, DataDistributionMetricsWorkload* self) {
+		loop {
+			state ReadYourWritesTransaction tr(cx);
+			state int i;
+			try {
+				for (i = 0; i < self->readPerTx; ++i)
+					wait(success(tr.get(self->keyForIndex(deterministicRandom()->randomInt(0, self->nodeCount))))); // read
+				for (i = 0; i < self->writePerTx; ++i)
+					tr.set(self->keyForIndex(deterministicRandom()->randomInt(0, self->nodeCount)), getRandomValue()); // write
+				wait(tr.commit());
+				++self->commits;
+			} catch (Error& e) {
+				wait(tr.onError(e));
+			}
+			tr.reset();
+		}
+	}
+
+	ACTOR Future<Void> resultConsistencyCheckClient(Database cx, DataDistributionMetricsWorkload* self) {
+		state Reference<ReadYourWritesTransaction> tr =
+		    Reference<ReadYourWritesTransaction>(new ReadYourWritesTransaction(cx));
+		loop {
+			try {
+				int startIndex = deterministicRandom()->randomInt(0, self->nodeCount - 1);
+				int endIndex = deterministicRandom()->randomInt(startIndex + 1, self->nodeCount);
+				state Key startKey = self->keyForIndex(startIndex);
+				state Key endKey = self->keyForIndex(endIndex);
+				// lastLessOrEqual
+				state KeySelector begin = KeySelectorRef(startKey.withPrefix(ddStatsRange.begin, startKey.arena()), true, 0);
+				state KeySelector end = KeySelectorRef(endKey.withPrefix(ddStatsRange.begin, endKey.arena()), false, 2);
+				Standalone<RangeResultRef> result = wait(tr->getRange(begin, end, GetRangeLimits(CLIENT_KNOBS->SHARD_COUNT_LIMIT)));
+				if (result.size() > 1) {
+					if (result[0].key <= begin.getKey() && result[1].key > begin.getKey()) {
+						TraceEvent(SevDebug, "DDMetricsConsistencyTest")
+						    .detail("Size", result.size())
+						    .detail("FirstKey", result[0].key.toString())
+						    .detail("SecondKey", result[1].key.toString())
+						    .detail("BeginKeySelector", begin.toString());
+					} else {
+						++self->errors;
+						TraceEvent(SevError, "TestFailure")
+						    .detail("Reason", "Result mismatches the given begin selector")
+						    .detail("Size", result.size())
+						    .detail("FirstKey", result[0].key.toString())
+						    .detail("SecondKey", result[1].key.toString())
+						    .detail("BeginKeySelector", begin.toString());
+					}
+					if (result[result.size()-1].key >= end.getKey() && result[result.size()-2].key < end.getKey()) {
```

Comment:
```suggestion
					if (result.readThroughEnd || (result[result.size()-1].key >= end.getKey() && result[result.size()-2].key < end.getKey())) {
```

Reply:
If `result.readThroughEnd = true`, it will throw `error_code_special_keys_cross_module_read` which is caught later. IMHO, this is okay?